src/node_report.cc

```diff
@@ -635,35 +714,47 @@ static void PrintVersionInformation(FILE* fp, Isolate* isolate) {
  * Function to print the JavaScript stack, if available
  *
  ******************************************************************************/
-static void PrintJavaScriptStack(FILE* fp, Isolate* isolate, DumpEvent event, const char* location) {
-  fprintf(fp, "\n================================================================================");
-  fprintf(fp, "\n==== JavaScript Stack Trace ====================================================\n\n");
+static void PrintJavaScriptStack(std::ostream& out, Isolate* isolate, DumpEvent event, const char* location) {
+  out << "\n================================================================================";
+  out << "\n==== JavaScript Stack Trace ====================================================\n\n";
 
 #ifdef _WIN32
   switch (event) {
   case kFatalError:
     // Stack trace on fatal error not supported on Windows
-    fprintf(fp, "No stack trace available\n");
+    out << "No stack trace available\n";
     break;
   default:
     // All other events, print the stack using StackTrace::StackTrace() and GetStackSample() APIs
-    PrintStackFromStackTrace(fp, isolate, event);
+    PrintStackFromStackTrace(out, isolate, event);
     break;
   }  // end switch(event)
 #else  // Unix, OSX
   switch (event) {
   case kException:
-  case kJavaScript:
+  case kJavaScript: {
     // Print the stack using Message::PrintCurrentStackTrace() API
-    Message::PrintCurrentStackTrace(isolate, fp);
+    std::FILE *stack_fp = std::tmpfile();
+    if( stack_fp != nullptr ) {
+      char stack_buf[64];
+      Message::PrintCurrentStackTrace(isolate, stack_fp);
+      std::fflush(stack_fp);
+      std::rewind(stack_fp);
+      while( std::fgets(stack_buf, sizeof(stack_buf), stack_fp) != nullptr ) {
+        out << stack_buf;
```

Comment:
It feels a bit painful to do this even in the case of when we are writing out to a file as it gets written to a file twice.  Having said that I don't have a good idea for how to avoid and be able to easily switch the output of the stream.

This might be something to put on our todo list in terms of working with V8 to get an API that can be used without having to write to a file.